rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2580 @@
+- New `create`, `get`, `update`, and `delete` methods will need to be made for Property Types, Data Types, and Link Types
+- The error conditions of the Type methods will be updated to include further validation errors that include but are not limited to:
+  - referencing a type that isn't in the embedding application
+  - submitting a type that is malformed
+
+# Reference-Level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Data Types
+
+A **Data Type** is a JSON schema that satisfies the following JSON meta-schema:
+
+```json
+{
+  "$id": "https://blockprotocol.org/type-system/0.2/schema/meta/data-type",
+  "description": "Specifies the structure of a Data Type",
+  "properties": {
+    "kind": {
+      "const": "dataType"
+    },
+    "$id": { "type": "string", "format": "uri" },
+    "name": { "type": "string" },
+    "description": { "type": "string" },
+    "type": { "type": "string" }
+  },
+  "required": ["kind", "$id", "name", "type"]
+}
+```
+
+### Primitive Top-Level Data Types
+
+This RFC defines the following primitive top-level Data Types
+
+- `Text`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/text",
+    "name": "Text",
+    "description": "An ordered sequence of characters",
+    "type": "string"
+  }
+  ```
+
+- `Number`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/number",
+    "name": "Number",
+    "description": "An arithmetical value (in the Real number system)",
+    "type": "number"
+  }
+  ```
+
+- `Boolean`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/boolean",
+    "name": "Boolean",
+    "description": "A True or False value",
+    "type": "boolean"
+  }
+  ```
+
+- `Null`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/null",
+    "name": "Null",
+    "description": "A placeholder value representing 'nothing'",
+    "type": "null"
+  }
+  ```
+
+- `Object`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/object",
+    "name": "Object",
+    "description": "A plain JSON object with no pre-defined structure",
+    "type": "object"
+  }
+  ```
+
+- `Empty List`
+
+  ```json
+  {
+    "kind": "dataType",
+    "$id": "https://blockprotocol.org/types/@blockprotocol/data-type/empty-list",
+    "name": "Empty List",
+    "description": "An Empty List",
+    "type": "array",
+    "maxItems": 0
+  }
+  ```
+
+### Creating new Data Types
+
+Due to the size of this RFC, creating new Data Types has been deemed out-of-scope and a [follow-up RFC](https://github.com/blockprotocol/blockprotocol/pull/355) is in the works to spec out how this could work, please consult it for plans on introducing a suite of Data Types including ones like `Date`, `Positive Number`, etc.:
+
```

Comment:
I think the reference-level explanation should largely be self-contained. It's mostly a rewrite of the guide-level in much more detail.

As such it's probably worth still having this disclaimer. We could perhaps consider removing it from the guide-level but I think it's better to answer a potential question earlier rather than to distract a reader who might go searching.